example/lib/src/examples/disk_file_store/disk_file_store.dart

```diff
@@ -0,0 +1,112 @@
+import 'dart:convert';
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:indexed_entity_store/indexed_entity_store.dart';
+
+typedef FileWithMetadata<T> = ({T metadata, String filepath});
+
+abstract class DiskFileConnector<
```

Comment:
Implementing this only as a connector runs into limitations as we can not ensure the "copy on write" and "delete on remove" behavior.

So the interface has to change to provide "file store", which then has specialized `read`/`write` methods.

Reply:
This has been solved by the store only exposing a subset of the methods, and taking care of the underlying file operations for each one.